primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java

```diff
@@ -235,9 +236,13 @@ protected Predicate createPredicate(FilterMeta filter, Field filterField,
                         ? cb.notEqual(fieldExpression, filterValueAsCollection.get().iterator().next())
                         : fieldExpression.in(filterValueAsCollection.get()).not();
             case BETWEEN:
-                throw new UnsupportedOperationException("MatchMode.BETWEEN currently not supported!");
+                Iterator<Object> iter1 = filterValueAsCollection.get().iterator();
+                return cb.and(cb.greaterThanOrEqualTo(fieldExpression, (Comparable) iter1.next()),
+                    cb.lessThanOrEqualTo(fieldExpression, (Comparable) iter1.next()));
```

Comment:
Inclusive on both sides based on the `isBetween` impl in `BetweenFilterConstraint`:
```java
    protected boolean isBetween(Comparable value, Comparable start, Comparable end) {
        return value.compareTo(start) >= 0 && value.compareTo(end) <= 0;
    }
```